src/JSInterop/Microsoft.JSInterop/src/Json/SimpleJson/SimpleJson.cs

```diff
@@ -2184,12 +2214,18 @@ public bool Remove(KeyValuePair<TKey, TValue> item)
 
                 public IEnumerator<KeyValuePair<TKey, TValue>> GetEnumerator()
                 {
-                    return _dictionary.GetEnumerator();
+                    lock (_lock)
+                    {
+                        return _dictionary.GetEnumerator();
+                    }
                 }
 
                 System.Collections.IEnumerator System.Collections.IEnumerable.GetEnumerator()
                 {
-                    return _dictionary.GetEnumerator();
+                    lock (_lock)
+                    {
+                        return _dictionary.GetEnumerator();
```

Comment:
This won't help, enumerator is being used outside the lock.

Reply:
Any cheap way of fixing this? I can create an enumerator wrapper and have it release the lock on Dispose but anything else seems overkill.

I'm not concerned too much about this as the callstacks that we have show another method is the one causing issues, but if there is something trivial (15 min) I'll do it.

Reply:
You can have it copy the underlying data. 

Is it feasible to eliminate this from the code? - ie does anything call this? If this is internal to simplejson can the calling code take the lock